### PR TITLE
Fix JS regex use to validate IRIS IAM strings

### PR DESCRIPTION
**Pull request by gregcorbett**

### CODE REVIEW

File: htdocs/web_portal/javascript/linking.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ function getRegExIdString() {
     // End with @iris.iam.ac.uk
     } else if (authType === "IRIS IAM - OIDC") {
         // var regExIdString = /^([a-f0-9]{8}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{12})@iris\-iam\.ac\.uk$/;
-        var regExIdString = /^.+@iris\-iam\.ac\.uk$/;
+        var regExIdString = /^.+@iris\-iam\.stfc\.ac\.uk\/$/
```
Comment (ineilson) [inline htdocs/web_portal/javascript/linking.js:88]: My regex is rusty but did you intend to force a trailing slash? Added line termination semi-colon.
```suggestion
        var regExIdString = /^.+@iris\-iam\.stfc\.ac\.uk$/;
```

Comment (gregcorbett) [reply htdocs/web_portal/javascript/linking.js:88]: I do mean to force a trailing slash, it's included in the ID string the IRIS IAM gives out. Good spot about the `;` :)

Comment (ineilson) [reply htdocs/web_portal/javascript/linking.js:88]: For a future change maybe but this very much feels like it shouldn't be hardcoded.

**Comment by ElliottKasoar** [comment]
Minor point but should `// End with @iris.iam.ac.uk` be updated to reflect the correct string?

It's probably obvious, but just to note that the commented out `regExIdString` above was planned to check the rest of the form of the ID string. It may be worth either implementing it for a more detailed check, or removing it if it's never going to be used (including it would "give away" the full form of the ID string, which may be a issue in itself).

**Comment by ineilson** [comment]
> Minor point but should `// End with @iris.iam.ac.uk` be updated to reflect the correct string?

Seems like a good idea.

**Comment by ineilson** [comment]
> It's probably obvious, but just to note that the commented out `regExIdString` above was planned to check the rest of the form of the ID string. It may be worth either implementing it for a more detailed check, or removing it if it's never going to be used (including it would "give away" the full form of the ID string, which may be a issue in itself).

Why was it not used in the first place, and what would be the implication of the 'give away'?

**Comment by ElliottKasoar** [comment]
> Why was it not used in the first place

From what I can recall, I think it was a combination of seeming unnecessary, and that I couldn't test it, as my instance of GOCDB didn't have IRIS IAM auth enabled (so I wasn't even registered myself).

I think the longer form was essentially a placeholder based on discussions with @gregcorbett about the expected full form of the ID string, but later we moved towards more minimal validation that was sufficient to differentiate the different auth types (e.g. `// End with @egi.eu` and `// Start with slash only`).


> and what would be the implication of the 'give away'?

I meant in the sense of helping people too much in guessing what an ID string should be that they're not in possession of, but I don't think it's really an issue?

**Comment by gregcorbett** [comment]
> Minor point but should // End with @iris.iam.ac.uk be updated to reflect the correct string?

Ah yes, I've changed the code but not the comment 🤦

**Comment by gregcorbett** [comment]
> > Minor point but should // End with @iris.iam.ac.uk be updated to reflect the correct string?
> 
> Ah yes, I've changed the code but not the comment 🤦

I've fixed this and rebased (into a single commit) the branch on top of the latest dev.

**Comment by gregcorbett** [comment]
This has been running on the IRIS instance since at least Jun 2022.

File: htdocs/web_portal/javascript/linking.js [hunk 2]
```diff
@@ -84,8 +84,8 @@ function getRegExIdString() {
 
     // End with @iris.iam.ac.uk
     } else if (authType === "IRIS IAM - OIDC") {
-        // var regExIdString = /^([a-f0-9]{8}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{12})@iris\-iam\.ac\.uk$/;
-        var regExIdString = /^.+@iris\-iam\.ac\.uk$/;
+        // var regExIdString = /^([a-f0-9]{8}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{4}\-[a-f0-9]{12})@iris\-iam\.stfc\.ac\.uk$/;
+        var regExIdString = /^.+@iris\-iam\.stfc\.ac\.uk\/$/;
```
Comment (Sae126V) [inline htdocs/web_portal/javascript/linking.js:88]: Just a clarification, Greg. So this now would be having / (relative URL) in the end?

Comment (gregcorbett) [reply htdocs/web_portal/javascript/linking.js:88]: Yep 😄

**Review by Sae126V — APPROVED** [review]
LGTM

File: htdocs/web_portal/javascript/linking.js (hunk 2 above)
Comment (gregcorbett) [reply htdocs/web_portal/javascript/linking.js:88]: it's a weridness of the IRIS IAM.

**Comment by gregcorbett** [comment]
rebasing on latest dev

**Pull request merged**